[CI] XPackRestIT test {p0=migrate/10_reindex/Test Reindex With Bad Data Stream Name} failing

**Build Scans:**
- [elasticsearch-periodic #5321 / release-tests](https://gradle-enterprise.elastic.co/s/jyrnzsmwci3pk)
- [elasticsearch-pull-request #45572 / release-tests](https://gradle-enterprise.elastic.co/s/j4i756bs3u4tm)
- [elasticsearch-pull-request #45528 / release-tests](https://gradle-enterprise.elastic.co/s/7ytjhywudkcpe)
- [elasticsearch-pull-request #45454 / release-tests](https://gradle-enterprise.elastic.co/s/opupzuedyh7xy)
- [elasticsearch-periodic #5309 / release-tests](https://gradle-enterprise.elastic.co/s/7zdy2lvjsycr4)
- [elasticsearch-periodic #5303 / release-tests](https://gradle-enterprise.elastic.co/s/hxdhhq46djmpa)
- [elasticsearch-pull-request #45409 / release-tests](https://gradle-enterprise.elastic.co/s/wjlncfil6euwi)
- [elasticsearch-periodic #5297 / release-tests](https://gradle-enterprise.elastic.co/s/3ljn5jyiywcno)
- [elasticsearch-periodic #5291 / release-tests](https://gradle-enterprise.elastic.co/s/yabxk4tgrhxmq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:yamlRestTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=migrate/10_reindex/Test Reindex With Nonexistent Data Stream}" -Dtests.seed=FFE1DC7DB5CD5155 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=tt-RU -Dtests.timezone=Europe/Astrakhan -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.test.rest.XPackRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dmigrate%2F10_reindex%2FTest%20Reindex%20With%20Bad%20Data%20Stream%20Name%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [migrate/10_reindex:22]: the error message was expected to match the provided regex but didn't
Expected: resource_not_found_exception
     but: was "no handler found for uri [/_migration/reindex?error_trace=true] and method [POST]"
```

**Issue Reasons:**
- [8.x] 9 consecutive failures in step release-tests
- [8.x] 9 failures in test test {p0=migrate/10_reindex/Test Reindex With Bad Data Stream Name} (3.5% fail rate in 259 executions)
- [8.x] 9 failures in step release-tests (100.0% fail rate in 9 executions)
- [8.x] 5 failures in pipeline elasticsearch-periodic (83.3% fail rate in 6 executions)
- [8.x] 4 failures in pipeline elasticsearch-pull-request (8.7% fail rate in 46 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -282,15 +282,6 @@ tests:
 - class: org.elasticsearch.packaging.test.ArchiveTests
   method: test60StartAndStop
   issue: https://github.com/elastic/elasticsearch/issues/118216
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  method: test {p0=migrate/10_reindex/Test Reindex With Bad Data Stream Name}
-  issue: https://github.com/elastic/elasticsearch/issues/118272
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  method: test {p0=migrate/10_reindex/Test Reindex With Unsupported Mode}
-  issue: https://github.com/elastic/elasticsearch/issues/118273
-- class: org.elasticsearch.xpack.test.rest.XPackRestIT
-  method: test {p0=migrate/10_reindex/Test Reindex With Nonexistent Data Stream}
-  issue: https://github.com/elastic/elasticsearch/issues/118274
 - class: org.elasticsearch.index.codec.vectors.es818.ES818HnswBinaryQuantizedVectorsFormatTests
   method: testSingleVectorCase
   issue: https://github.com/elastic/elasticsearch/issues/118306

--- a/test/test-clusters/src/main/java/org/elasticsearch/test/cluster/FeatureFlag.java
+++ b/test/test-clusters/src/main/java/org/elasticsearch/test/cluster/FeatureFlag.java
@@ -19,7 +19,8 @@ public enum FeatureFlag {
     TIME_SERIES_MODE("es.index_mode_feature_flag_registered=true", Version.fromString("8.0.0"), null),
     FAILURE_STORE_ENABLED("es.failure_store_feature_flag_enabled=true", Version.fromString("8.12.0"), null),
     SUB_OBJECTS_AUTO_ENABLED("es.sub_objects_auto_feature_flag_enabled=true", Version.fromString("8.16.0"), null),
-    INFERENCE_UNIFIED_API_ENABLED("es.inference_unified_feature_flag_enabled=true", Version.fromString("8.18.0"), null);
+    INFERENCE_UNIFIED_API_ENABLED("es.inference_unified_feature_flag_enabled=true", Version.fromString("8.18.0"), null),
+    MIGRATION_REINDEX_ENABLED("es.reindex_data_stream_feature_flag_enabled=true", Version.fromString("8.18.0"), null);
 
     public final String systemProperty;
     public final Version from;

